source/contracts/Augur.sol

```diff
@@ -3,13 +3,98 @@ pragma solidity 0.4.17;
+    event UniverseCreated(address indexed parentUniverse, address indexed childUniverse);
+
     function trustedTransfer(ERC20 _token, address _from, address _to, uint256 _amount) public onlyWhitelistedCallers returns (bool) {
         require(_amount > 0);
         _token.transferFrom(_from, _to, _amount);
         return true;
     }
+
+    //
+    // Logging
+    //
+
+    function logMarketCreated(IUniverse _universe, address _market, address _marketCreator, uint256 _marketCreationFee, string _extraInfo) public returns (bool) {
+        require(_universe.isContainerForReportingWindow(IReportingWindow(msg.sender)));
+        MarketCreated(_universe, _market, _marketCreator, _marketCreationFee, _extraInfo);
+        return true;
+    }
+
+    function logDesignatedReportSubmitted(IUniverse _universe, address _reporter, address _market, address _stakeToken, uint256 _amountStaked, uint256[] _payoutNumerators) public returns (bool) {
+        require(_universe.isContainerForStakeToken(IStakeToken(msg.sender)));
+        DesignatedReportSubmitted(_universe, _reporter, _market, _stakeToken, _amountStaked, _payoutNumerators);
+        return true;
+    }
+
+    function logReportSubmitted(IUniverse _universe, address _reporter, address _market, address _stakeToken, uint256 _amountStaked, uint256[] _payoutNumerators) public returns (bool) {
+        require(_universe.isContainerForStakeToken(IStakeToken(msg.sender)));
+        ReportSubmitted(_universe, _reporter, _market, _stakeToken, _amountStaked, _payoutNumerators);
+        return true;
+    }
+
+    function logWinningTokensRedeemed(IUniverse _universe, address _reporter, address _market, address _stakeToken, uint256 _amountRedeemed, uint256 _reportingFeesReceived, uint256[] _payoutNumerators) public returns (bool) {
+        require(_universe.isContainerForStakeToken(IStakeToken(msg.sender)));
+        WinningTokensRedeemed(_universe, _reporter, _market, _stakeToken, _amountRedeemed, _reportingFeesReceived, _payoutNumerators);
+        return true;
+    }
+
+    function logReportsDisputed(IUniverse _universe, address _disputer, address _market, uint8 _reportingPhase, uint256 _disputeBondAmount) public returns (bool) {
+        require(_universe.isContainerForMarket(IMarket(msg.sender)));
+        ReportsDisputed(_universe, _disputer, _market, _reportingPhase, _disputeBondAmount);
+        return true;
+    }
+
+    function logMarketFinalized(IUniverse _universe, address _market) public returns (bool) {
+        require(_universe.isContainerForMarket(IMarket(msg.sender)));
+        MarketFinalized(_universe, _market);
+        return true;
+    }
+
+    function logOrderCanceled(IUniverse _universe, address _shareToken, address _sender, bytes32 _orderId, uint8 _orderType, uint256 _tokenRefund, uint256 _sharesRefund) public onlyWhitelistedCallers returns (bool) {
```

Comment:
Bleh, I'm not a fan of mixing trust strategies here, though I suppose there isn't really any alternative option?  I'm referring to how some of these functions use `onlyWhitelistedCallers` and others use `require(universe.isContainerFor...)`.  😢

Reply:
Yea, the trading contracts are centralized and whitelisted so it's a bit different.